feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c

```diff
@@ -103,6 +101,56 @@ void get_channel_bandwidth(const char* htmode, int *channel_bandwidth)
 	else if(!strcmp(htmode, "HT80"))
 		*channel_bandwidth=80;
 }
+/*
+ * A simple average is calculated against the Noise floor samples.
+ * - Returns a zero, if there are not enough samples in the list.
+ * - Or, returns a calculated avg of the noise samples.
+ * - A circular buffer is considered for storing the samples. New elements replace the
+ * old ones in this buffer.
+ * - The number of samples required is an integer value calculated based on the
+ * configuration time and the sample time.
+ * Example: If the config_time is 120 sec, and sample time is 15 sec, then number of
+ * samples required for averaging is 120/15 = 8 samples.
+ */
+int rrm_calculate_avg_noise_floor(rrm_entry_t *rrm_data, int nf, int config_time, int sample_time)
+{
+	int ii;
+	double avg_mW = 0;
+	int num_samples = config_time/sample_time;
+
+	if (num_samples >= RRM_MAX_NF_SAMPLES)
+		num_samples = RRM_MAX_NF_SAMPLES;
+
+	/* Replace the oldest element in the list with the new element */
+	rrm_data->rrm_chan_nf_samples[rrm_data->rrm_chan_nf_next_el] = nf;
+
+	/* Update the index to the oldest element index taking care of the boundary */
+	rrm_data->rrm_chan_nf_next_el = (rrm_data->rrm_chan_nf_next_el+1)%num_samples;
+
+	if (rrm_data->rrm_chan_nf_num_el < num_samples)
+	{
+		rrm_data->rrm_chan_nf_num_el++;
+		return 0;
+	}
+
+	/*
+	 * Convert dBm to milliWatts, calculate average,
+	 * convert the averaged milliWats back to dBm
+	 */
+	for (ii = 0; ii < num_samples; ii++)
+	{
+		avg_mW += dBm_to_mwatts(rrm_data->rrm_chan_nf_samples[ii]);
+	}
+	avg_mW = avg_mW/num_samples;
+
+	return ((int)(mWatts_to_dBm(avg_mW)));
```

Comment:
And this would be unchanged.

Or we could convert the threshold 1 time and just make the decision in mW instead of dBm

Reply:
I would vote for leaving this as it is. This way the input and output of this function stay consistent (dBm).

Reply:
Makes sense.